sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java

```diff
@@ -316,6 +316,35 @@ static boolean removeUnsupportedDataTypes(@NonNull Map<String, Object> data) {
         return removed;
     }
 
+    /**
+     * Truncates the provided stack trace to the specified limit per line and returns the truncated stack trace.
+     *
+     * @param stackTrace the stack trace to truncate
+     * @param maxStackTraceLineLength the maximum length of each line in the stack trace
+     * @param tag the tag to use in logs
+     * @param L the logger
+     * @return the truncated stack trace
+     */
+    protected static String applyInternalLimitsToStackTraces(@NonNull String stackTrace, final int maxStackTraceLineLength, @NonNull String tag, @NonNull ModuleLog L) {
+        assert stackTrace != null;
+        assert maxStackTraceLineLength >= 1;
+        assert tag != null;
+        assert L != null;
+        
+        StringBuilder sb = new StringBuilder(stackTrace.length());
+
+        String[] stackTraceLines = stackTrace.split("\n");
+        for (int i = 0; i < stackTraceLines.length; i++) {
+            String truncatedLine = UtilsInternalLimits.truncateString(stackTraceLines[i], maxStackTraceLineLength, L, tag);
+            if (i != 0) {
+                sb.append("\n");
```

Comment:
if you append this first in the loop wouldnt it make the last line not end with new line?

Reply:
yes that is the purpose, last line shouldn't have a newline right?

Reply:
🤷